src/bundle/Command/MigrateNamespacesCommand.php

```diff
@@ -0,0 +1,168 @@
+        PermissionResolver $permissionResolver,
+        UserService $userService,
+        Gateway $gateway
+    ) {
+        $this->objectCount = null;
+        parent::__construct(null, $permissionResolver, $userService);
+        $this->gateway = $gateway;
+    }
+
+    public function configure(): void
+    {
+        parent::configure();
+
+        $this
+            ->setName(self::$defaultName)
+            ->addOption(
+                'cursor-start',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Internal option - only used for subprocesses',
+            )
+            ->addOption(
+                'cursor-stop',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Internal option - only used for subprocesses',
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->cursorStart = $input->getOption('cursor-start') !== null ? (int) $input->getOption('cursor-start') : null;
+        $this->cursorStop = $input->getOption('cursor-stop') !== null ? (int) $input->getOption('cursor-stop') : null;
+
+        // Check that both --cursor-start and cursor-start are set, or neither
+        if (($this->cursorStart === null) xor ($this->cursorStop === null)) {
+            throw new RuntimeException('The options --cursor-start and -cursor-stop are only for internal use !');
+        }
+
+        parent::execute($input, $output);
+
+        return self::SUCCESS;
+    }
+
+    protected function getObjectCount(): int
+    {
+        if ($this->objectCount === null) {
+            $this->output->writeln('Fetching number of objects to process. This might take several minutes if you have many records in ezcontentobject_attribute table.');
+            $this->objectCount = $this->gateway->countRichtextAttributes();
+        }
+
+        return $this->objectCount;
+    }
+
+    protected function iterate(): void
+    {
+        $limit = $this->getIterationCount();
+        $cursor = [
+            'start' => -1,
+            'stop' => null,
+        ];
+
+        $contentAttributeIDs = $this->gateway->getContentObjectAttributeIds($cursor['start'], $limit);
+        $cursor['stop'] = $this->getNextCursor($contentAttributeIDs);
+        while ($cursor['stop'] !== null) {
+            $this->createChildProcess($cursor, count($contentAttributeIDs));
+
+            $cursor['start'] = $cursor['stop'];
+            $contentAttributeIDs = $this->gateway->getContentObjectAttributeIds($cursor['start'], $limit);
+            $cursor['stop'] = $this->getNextCursor($contentAttributeIDs);
+        }
+    }
+
+    protected function completed(): void
+    {
+        $this->output->writeln(PHP_EOL . 'Completed');
+    }
+
+    protected function getNextCursor(array $contentAttributeIDs): ?int
+    {
+        $lastId = count($contentAttributeIDs) > 0 ? end($contentAttributeIDs)['id'] : null;
+
+        return $lastId;
```

Comment:
There's no need to allocate extra memory for `$lastId`. You should return it immediately:
```suggestion
        return count($contentAttributeIDs) > 0 ? end($contentAttributeIDs)['id'] : null;
```